shared/src/main/scala/ListHelpers.scala

```diff
@@ -39,6 +39,23 @@ object ListHelpers:
 
   end filter
 
+  def interleave(left: VList, right: VList)(using ctx: Context): VList =
+    val out = ArrayBuffer.empty[VAny]
+    var mutLeft = left.toList
+    var mutRight = right.toList
+    while mutLeft.nonEmpty && mutRight.nonEmpty do
+      out += mutLeft.head
+      out += mutRight.head
+      mutLeft = mutLeft.tail
+      mutRight = mutRight.tail
+
+    out ++= mutLeft
+    out ++= mutRight
```

Comment:
```suggestion
    val leftIter = left.iterator
    val rightIter = right.iterator
    while leftIter.hasNext && rightIter.hasNext do
      out += leftIter.next()
      out += rightIter.next()

    out ++= leftIter
    out ++= rightIter
```
I'm not sure if those last two lines will still work with an iterator, but they can definitely be modified a bit to work

Reply:
Once again, I blame copilot